examples/instance_kind/model.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import numpy as np
+import torch
+import triton_python_backend_utils as pb_utils
+from torch.utils.dlpack import to_dlpack
+
+
+class TritonPythonModel:
+
+    def initialize(self, args):
+        """
+        This function initializes pre-trained ResNet50 model,
+        depending on the value specified by an `instance_group` parameter
+        in `config.pbtxt`. 
+
+        Depending on what `instance_group` was specified in 
+        the config.pbtxt file (KIND_CPU, KIND_GPU, or both), the model instance
```

Comment:
KIND_CPU or KIND_GPU. remove `or both`.
This initialize function is called per model instance. One model instance will only carry KIND_CPU, KIND_GPU or KIND_MODEL. See here: https://github.com/triton-inference-server/common/blob/main/protobuf/model_config.proto#L145-L182

Reply:
Also in line 47 we are only setting the device to "cuda" if the kind is GPU. otherwise to cpu. Hence, it is not both.

Reply:
I was referring to the option of loading model instances, one on CPU and other on GPU. 
If 
```
instance_group [{ kind: KIND_CPU }, { kind: KIND_GPU}]
```
is present in the config, then one instance will be loaded to cpu, another instance to gpu. I'll remove `or both` to avoid confusion.

Reply:
The initialize function is under scope of the model instance. Hence, or both will confuse users.

Reply:
Thanks for rising this concern, agree with this